test/TestProjects/AmbiguousMethods-LowLevel/AmbiguousClient.cs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace AmbiguousMethods_LowLevel
+{
+    /// <summary> Existing Ambiguous service client. Methods are changed to intentionally have ambiguity with DPG. </summary>
+    public partial class AmbiguousClient
+    {
+        private readonly HttpPipeline pipeline = HttpPipelineBuilder.Build(ClientOptions.Default);
+
+        /// <param name="cancellationToken"> CancellationToken to use. </param>
+        public virtual async Task<Response> NoParamAsync(CancellationToken cancellationToken = default)
+        {
+            return await pipeline.ProcessMessageAsync(null, null).ConfigureAwait(false);
+        }
+
+        /// <param name="a"> The string to use. </param>
+        /// <param name="cancellationToken"> CancellationToken to use. </param>
+        public virtual Response NoParam(string a = null, CancellationToken cancellationToken = default)
+        {
+            return pipeline.ProcessMessage(null, null);
+        }
+
+        /// <param name="optionalParamA"> The String to use. </param>
+        /// <param name="cancellationToken"> CancellationToken to use. </param>
+        public virtual async Task<Response> OneOptionalParamAsync(string optionalParamA = null, CancellationToken cancellationToken = default)
+        {
+            return await pipeline.ProcessMessageAsync(null, null).ConfigureAwait(false);
+        }
+
+        /// <param name="a"> The string to use. </param>
+        /// <param name="optionalParamA"> The String to use. </param>
+        /// <param name="cancellationToken"> CancellationToken to use. </param>
+        public virtual Response OneOptionalParam(string a = null, string optionalParamA = null, CancellationToken cancellationToken = default)
+        {
+            return pipeline.ProcessMessage(null, null);
+        }
+
+        /// <param name="optionalParamB"> The Integer to use. </param>
+        /// <param name="cancellationToken"> CancellationToken to use. </param>
+        public virtual async Task<Response> TwoOptionalParamAsync(int? optionalParamB = null, CancellationToken cancellationToken = default)
```

Comment:
Similarly, shouldn't the two optional parameters method take two optional parameters?  I realize we can't change the method name, but could we add a comment to describe what we're validating with this method?

Reply:
sure.